### PR TITLE
Bug#1313910 fixes - added information on nodePorts

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://bugzilla.redhat.com/show_bug.cgi?id=1313910
Added example for configuring service nodePort

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
@TheDiemer 
@zhouying7780 

Can you please review this?

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 1]
```diff
@@ -430,7 +430,7 @@ endif::[]
 == Using a Service NodePort
 
 Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster.
-NodePorts are in the 30-60k range by default, which means a NodePort is
+NodePorts are in the 30k-60k range by default, which means a NodePort is
```
Comment (pecameron) [inline dev_guide/getting_traffic_into_cluster.adoc]: The range is used when the nodeport is not specified. The user can specify the desired nodeport (or just use the default)

**Review by pecameron — COMMENTED** [review]
/lgtm

**Comment by gaurav-nelson** [comment]
@smarterclayton Can you please check this one? and also comment on the conversation thread at https://bugzilla.redhat.com/show_bug.cgi?id=1313910

**Review by pecameron — APPROVED** [review]
You should add that you may need a firewall rule to access to the nodeport.

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 2]
```diff
@@ -429,17 +429,85 @@ endif::[]
...
-unlikely to match a service's intended port (for example, 8080 may be exposed
-as 31020). This use of ports is wasteful of scarce port resources.
-However, it is slightly easier to set up. Again, this requires more privileges.
+Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster. You can specify the desired nodePort. When a nodePort is not specified, it is set to a
+port number between `30000` - `60000` by default. Therefore, a NodePort is
+unlikely to match a service's intended port (for example, `8080` may be exposed
+as `31020`). This use of ports is wasteful of scarce port resources. However, it
+is slightly easier to set up. Again, this requires more privileges.
 
 The administrator must ensure the external IPs are routed to the nodes and local
```
Comment (gaurav-nelson) [inline dev_guide/getting_traffic_into_cluster.adoc]: @pecameron Firewall rules line is listed here already.

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 3]
```diff
@@ -429,17 +429,85 @@ endif::[]
...
 
-Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster.
-NodePorts are in the 30-60k range by default, which means a NodePort is
-unlikely to match a service's intended port (for example, 8080 may be exposed
-as 31020). This use of ports is wasteful of scarce port resources.
-However, it is slightly easier to set up. Again, this requires more privileges.
+Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster. You can specify the desired nodePort. When a nodePort is not specified, it is set to a
+port number between `30000` - `60000` by default. Therefore, a NodePort is
+unlikely to match a service's intended port (for example, `8080` may be exposed
+as `31020`). This use of ports is wasteful of scarce port resources. However, it
```
Comment (smarterclayton) [inline dev_guide/getting_traffic_into_cluster.adoc]: I wouldn't call this wasteful.  Also, the default node port range in OpenShift is 30000-32000 (but admins can widen it).

I would say:

> NodePorts are intended to allow load balancing across an entire cluster and are used by the LoadBalancer service type when running on a cloud provider.  You may specify a specific nodePort (if not already in use) or allow the system to assign you one.  When the port is assigned, all nodes in the cluster will allow incoming traffic to that port and then direct the traffic from that node to a local or remote endpoint under the service.  By default, OpenShift allocates ports between 30000 and 32000 as NodePorts.  For very high density environments, you may need to increase the range of ports.  OpenShift automatically opens the node port range at install time for firewalls and cloud security groups for new clusters.

Comment (pecameron) [reply dev_guide/getting_traffic_into_cluster.adoc]: @smarterclayton how do you increase the range of ports? Nice description.

Comment (gaurav-nelson) [reply dev_guide/getting_traffic_into_cluster.adoc]: Thanks @smarterclayton for the updated description. Yes the process for increasing range of ports will make a good addition to the docs.

**Comment by gaurav-nelson** [comment]
@openshift/team-documentation peer review please

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 4]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
-NodePorts are in the 30-60k range by default, which means a NodePort is
-unlikely to match a service's intended port (for example, 8080 may be exposed
-as 31020). This use of ports is wasteful of scarce port resources.
-However, it is slightly easier to set up. Again, this requires more privileges.
+Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster. NodePorts are intended to allow load balancing across an entire cluster and are used by the LoadBalancer service type when running on a cloud provider. You can specify a specific nodePort (if not already in use) or allow the system to assign you one. When the nodePort is assigned, all nodes in the cluster will allow incoming traffic to that port and then direct the traffic from that port to a local or remote endpoint under the service. By default, {product-title} allocates ports between `30000` and `32000` as NodePorts. For very high density environments, you may need to increase the range of ports. OpenShift automatically opens the node port range at install time for firewalls and cloud security groups for new clusters.
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: s/NodePorts are intended to allow/NodePorts allow
port, and then direct
s/OpenShift/{product-title}
s/node port range/nodePort range
s/at install time/during installation

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 5]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
 
 The administrator must ensure the external IPs are routed to the nodes and local
 firewall rules on all nodes allow access to the open port.
 
 NodePorts and externalIP are independent and both can be used concurrently.
 
+[NOTE]
+====
+When enabled, a service NodePort listens on all nodes in the cluster. The service usually has multiple endpoints if there are more than one replicas (Domain Controller or Replication Controller) running. If traffic can get to any node in the cluster, the nodePort connects and traffic goes to one of the endpoints.
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: s/Domain Controller or Replication Controller/domain controller or replication controller

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 6]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
+====
+
+
+[[example-connecting-mysql-using-nodeport]]
+.Example Connecting to MySQL outside OpenShift using NodePort
+
+. Create a new database pod with MySQL:
++
+
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: remove extra line between code block start and + to keep everything in line

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 7]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
+[source,bash]
+----
+$   oc new-app mysql --env=MYSQL_USER=user \
+--env=MYSQL_PASSWORD=pass --env=MYSQL_DATABASE=testdb -l db=mysql
+----
+. Edit the service object definition to include nodePort:
++
+
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: same here

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 8]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
+[source,bash]
+----
+$ oc get svc mysql -o yaml |grep nodePort
+    nodePort: 30234
+----
+
+. Verify service access on all nodes:
++
+
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: extra line here too

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 9]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
+    deploymentconfig: "mysql"
+  type: "NodePort"
+  sessionAffinity: "None"
+status:
+  ...
+----
+
+. Get the automatically assigned nodePort numder:
++
+
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: remove extra line here too

File: dev_guide/getting_traffic_into_cluster.adoc [hunk 10]
```diff
@@ -429,17 +429,81 @@ endif::[]
...
+[[example-connecting-mysql-using-nodeport]]
+.Example Connecting to MySQL outside OpenShift using NodePort
+
+. Create a new database pod with MySQL:
++
+
+[source,bash]
+----
+$   oc new-app mysql --env=MYSQL_USER=user \
```
Comment (ahardin-rh) [inline dev_guide/getting_traffic_into_cluster.adoc]: looks like extra spacing between $ and the start of the command

**Comment by ahardin-rh** [comment]
@gaurav-nelson A few suggestions. Otherwise, looks good! 🍻

**Comment by gaurav-nelson** [comment]
Added and example and done minor changes to the existing content. No revision history required.

**Comment by gaurav-nelson** [comment]
Thanks @ahardin-rh for your comments, I addressed them all in the latest commit.

**Pull request merged**